src/core/task.h

```diff
@@ -0,0 +1,51 @@
+#ifndef CORE_TASK_H
+#define CORE_TASK_H
+
+#include <stdint-gcc.h>
+/// status of task
+typedef enum{
+	TASK_ACTIVE, ///< task is currently running
+	TASK_ACTIVE_WTM,  ///< task is currently active but it is waiting on motion
+	TASK_READY, ///< task is ready
+	TASK_DONE, ///< finished task
+	TASK_BLOCKED ///< task is blocked currently blocked because of some obstacle
+} task_states_t ;
+
+/// types of task.
+typedef enum{
+	TASK_USER, ///< user tasks (collect balls task ... )
+	TASK_SYSTEM_PREEXECUTE,  /*!<  system tasks are tasks that have to be updated every cycle ( set glovbal time, update led, ... )
+	 	 	 	 	 	 	 	 Tasks with this type will be executed before user tasks */
+	TASK_SYSTEM_POSTEXECUTE, ///< same as TASK_SYSTEM_PREEXECUTE but this tasks will be executed after user task.
+	TASK_BACKPROC ///< this task will be run while robot is in motion wait state
```

Comment:
Just a few questions about this one.

When are these exactly supposed to run? Are tasks of type `TASK_BACKPROC` those that should be run while, for example, the robot stopped due to encountering an enemy robot? 

What would an example of such a task be?